addons/mail/static/src/discuss/core/web/discuss_core_web_service.js

```diff
@@ -50,11 +50,18 @@ export class DiscussCoreWeb {
             ({ detail: { channel, message } }) => {
                 if (
                     !this.ui.isSmall &&
-                    channel.correspondent !== this.store.odoobot &&
+                    !(
+                        channel.correspondent === this.store.odoobot && this.store.odoobotOnboarding
+                    ) &&
                     !message.isSelfAuthored
                 ) {
                     this.chatWindowService.insert({ thread: channel });
                 }
+                if (channel.correspondent === this.store.odoobot && this.store.odoobotOnboarding) {
+                    // odoobot onboarding was cancelled
+                    // this lets new messages of odoobot be managed as any other new messages
+                    this.store.odoobotOnboarding = false;
+                }
```

Comment:
not so fond of repeating the check on `channel.correspondent === this.store.odoobot && this.store.odoobotOnboarding`
the negated conjunction is also a bit difficult to grasp, not to mention that the formatting is not great
I think the readability could be improved:
```suggestion
                if (this.ui.isSmall || message.isSelfAuthored) {
                    return;
                }
                if (this.store.odoobotOnboarding && channel.correspondent === this.store.odoobot) {
                    // don't open chat window only for the first message of
                    // odoobot onboarding
                    this.store.odoobotOnboarding = false;
                    return;
                }
                this.chatWindowService.insert({ thread: channel });
```

Reply:
Much better indeed. I changed it, thanks!